### ISSUE TITLE
Remove current icons system and switch to emoji

### ISSUE THREAD

**Issue by DarthGandalf** [kvirc/KVIrc#1475]
**Reported by wodim on 24 Jan 2015 22:48:51 UTC**
This is from my personal wishlist:
- Remove the current icons system. In its place, use Ctrl+I to show italics, as mIRC does.
- Use emoji instead. These icons would work only in UTF-8 networks, but they are future-proof, widely known, and already work on several IRC clients (most of them, for smartphones and tablets).

---

Migrated from: https://svn.kvirc.de/kvirc/ticket/1475

**Comment by DarthGandalf** [comment]
**Comment by DarthGandalf on 24 Jan 2015 23:24:12**
But there must be a setting to turn it off.
Now I remove all the icons from filesystem and config to make them disappear.

**Comment by HelLViS69** [comment]
Duplicate of #1525

**Comment by DarthGandalf** [comment]
Why do you think it's duplicate? #1525 stays within current icon system. Emoji are a special block in Unicode.

**Comment by un1versal** [comment]
+1 for this, the current texticons is somewhat broken.

Im not sure how implementing emoji via font is going to work across systems that may not have a specific font that supports this,

I had briefly discussed with pragma, another **somewhat** related font issue, a specific font set in a theme may not be available in another OS + add emoji support and this suddenly looking like a very hard job to implement.

**Comment by wodim** [comment]
What systems don't support emoji? Broken or very outdated ones? 💁🏿

**Comment by un1versal** [comment]
Not all OS carry same fonts and not all fonts support emoji

**Comment by staticfox** [comment]
Ones that _obviously_ need a hardware change :)

**Comment by un1versal** [comment]
> Ones that obviously need a hardware change :)

still banging on this door? door was taken down and recycled so now its a brick wall.

**Comment by ekolis** [comment]
So no typing :) to get a smiley - you'd have to paste an actual ☺ character in, or choose it from a menu?

**Comment by ekolis** [comment]
Notice how the ☺ I pasted in isn't even recognizable in this font as a smiley - it could just as easily be a frownie or a "meh" face!

**Comment by AndrioCelos** [comment]
Did it ever work that way?

**Comment by ekolis** [comment]
No, typing :) never gave me a graphical smiley. I'm saying I'd much rather have that than have to pick one from a menu. But the menu should be implemented as well, for emoji without simple text representations.

**Comment by AndrioCelos** [comment]
Seems like that could be done with a script somewhat easily.

**Comment by ekolis** [comment]
Does such a script exist? I don't know how to create them...

**Comment by wodim** [comment]
Inputting emoji is a task that should be left to the OS. Unfortunately only OS X lets users type emojis easily...

**Comment by un1versal** [comment]
> Does such a script exist? I don't know how to create them...

Not for that exactly but with a little imagination its possible to modify to use for these smileys.

https://gist.github.com/un1versal/12c102230d4c5774c10304527c642047

**Comment by ekolis** [comment]
Oh, I see that emoji replacement was added recently after all! But there is one issue - emoji should only be replaced if they are "whole words". For instance, if I type >:D to get an evil grinning face, I instead get > followed by a regular grinning face, since the evil grinning face isn't supported, which is confusing. It would be better to not generate an emoji at all in that case!
